src/shared/has-own.js

```diff
@@ -0,0 +1,12 @@
+/**
+ * Polyfill for `Object.hasOwn`.
+ *
+ * `hasOwn(someObject, property)` should be used instead of
+ * `someObject.hasOwnProperty(name)`.
+ *
+ * @param {object} object
+ * @param {string} property
+ */
+export function hasOwn(object, property) {
+  return Object.prototype.hasOwnProperty.call(object, property);
+}
```

Comment:
We do have a dependency on the core-js project which has a polyfill for `Object.hasOwn(...)`. However core-js's implementation has a lot of layers of indirection, which I don't like. Also it is preferable in the client to use polyfills in a way that doesn't pollute the global environment, in order to avoid conflicts with the host page. I am tempted to suggest that we drop core-js entirely and, when we need polyfills in future create a module that either:

1. Includes a simple implementation, as in this case OR
2. Imports an existing implementation from a third-party package if more complex, eg. from https://github.com/github/browser-support